src/EnergyPlus/DataHeatBalance.cc

```diff
@@ -1294,71 +1294,62 @@ Real64 ComputeNominalUwithConvCoeffs(EnergyPlusData &state,
     //      Interior horizontal surfaces, heat flow up     IP: 0.61  SI: 0.1074
     //      Interior horizontal surfaces, heat flow down   IP: 0.92  SI: 0.1620
     //      Interior vertical surfaces                     IP: 0.68  SI: 0.1198
-    // This section shows the same value in 90.1-2007 and 90.2-2010
-
-    // Using/Aliasing
-    using DataSurfaces::ExternalEnvironment;
-    using DataSurfaces::Ground;
-    using DataSurfaces::GroundFCfactorMethod;
-    using DataSurfaces::SurfaceClass;
+    // This section shows the same value in 90.1-2010 and 90.2-2010
+    // Note that this report does not use the semi-exterior surface value because
+    // EnergyPlus does not have a way to specifically tell whether or not a surface
+    // is connected to a semi-exterior area of the building.  Users can always use
+    // the Nominal U-Value to manually calculated this.  The values calculated here
+    // are simply reported to the EIO file and not used for any calculations.
 
     // Return value
     Real64 NominalUwithConvCoeffs; // return value
 
+    static constexpr std::array<Real64, static_cast<int>(DataSurfaces::SurfaceClass::Num)> filmCoefs = {
+        0.0,       // None
+        0.1197548, // Wall
+        0.1620212, // Floor
+        0.1074271, // Roof
+        0.0,       // IntMass
+        0.0,       // Detached_B
+        0.0,       // Detached_F
+        0.1197584, // Window
+        0.1197584, // GlassDoor
+        0.1197584, // Door
+        0.0,       // Shading
+        0.0,       // Overhang
+        0.0,       // Fin
+        0.0,       // TDD_Dome
+        0.0        // TDD_Diffuser
+    };
```

Comment:
These coefficients are numerically correct now and same as before. The array lines up with enum class SurfaceClass in DataSurfaces.hh

Reply:
For some reason, even though your comment was positive, it jogged my memory that the coefficient that had to be fixed shows up three other times in this same statement. 🤦‍♂️  Just committed that fix.  Trying to juggle too many branches at once...

Reply:
Wow, I looked right at those, and yes they are different in the last 2 decimal places. Reviewing code sure can be difficult at times. OK, so now my comment is accuracte. Thanks for catching those.

Reply:
Using the array is nice and tidy, but it comes with the risk of misalignment if the surface class list is rearranged or added to in the future. A comment by the enum delcaration referring to this array is in order to alert future developers. I haven't examined the unit test carefully, but does it cover every enum case and compare against a hard value in the unit test so that it will fail if there's any misalignment?

Reply:
try static_assert.